src/main/java/software/amazon/awssdk/crt/http/Http2ClientConnection.java

```diff
@@ -0,0 +1,176 @@
+     */
+    public CompletableFuture<Void> changeSettings(final Http2ConnectionSetting settings[]) {
+        throw new CrtRuntimeException("Unimplemented");
+    }
+
+    /**
+     * Send a PING frame. Round-trip-time is calculated when PING ACK is received
+     * from peer.
+     *
+     * @param pingData 8 Bytes data with the PING frame or null for not include data in ping
+     *
+     * @return When this future completes without exception, the peer has
+     *         acknowledged the PING and future will be completed with the round
+     *         trip time in nano seconds for the connection.
+     */
+    public CompletableFuture<Long> sendPing(final byte[] pingData) {
+        CompletableFuture<Long> completionFuture = new CompletableFuture<>();
+        http2ClientConnectionSendPing(getNativeHandle(), completionFuture, pingData);
+        return completionFuture;
+    }
+
+    public CompletableFuture<Long> sendPing() {
+        CompletableFuture<Long> completionFuture = new CompletableFuture<>();
+        http2ClientConnectionSendPing(getNativeHandle(), completionFuture, null);
+        return completionFuture;
+    }
+
+    /**
+     * Send a custom GOAWAY frame.
+     *
+     * Note that the connection automatically attempts to send a GOAWAY during
+     * shutdown (unless a GOAWAY with a valid Last-Stream-ID has already been sent).
+     *
+     * This call can be used to gracefully warn the peer of an impending shutdown
+     * (http2_error=0, allow_more_streams=true), or to customize the final GOAWAY
+     * frame that is sent by this connection.
+     *
+     * The other end may not receive the goaway, if the connection already closed.
+     *
+     * @param Http2ErrorCode   The HTTP/2 error code (RFC-7540 section 7) to send.
+     *                         `enum Http2ErrorCode` lists official codes.
+     * @param allowMoreStreams If true, new peer-initiated streams will continue to
+     *                         be acknowledged and the GOAWAY's Last-Stream-ID will
+     *                         be set to a max value. If false, new peer-initiated
+     *                         streams will be ignored and the GOAWAY's
+     *                         Last-Stream-ID will be set to the latest acknowledged
+     *                         stream.
+     * @param debugData        Optional debug data to send. Size must not exceed
+     *                         16KB. null is acceptable to not include debug data.
+     */
+    public void sendGoAway(final Http2ErrorCode Http2ErrorCode, final boolean allowMoreStreams,
+            final byte[] debugData) {
+        throw new CrtRuntimeException("Unimplemented");
+    }
+
+    public void sendGoAway(final Http2ErrorCode Http2ErrorCode, final boolean allowMoreStreams) {
+        throw new CrtRuntimeException("Unimplemented");
+    }
+
+    /**
+     * Increment the connection's flow-control window to keep data flowing.
```

Comment:
In the manual management mode, how does the user know whenever the window size shrinks so they can call this method?

Reply:
that's a good callout! I'll need to think more about it. We currently don't have any feedback when windows shrinks, user will need to track the window themselves, which may not be a good option, especially the connection manager will handle all the connections.